cli/command/container/restart.go

```diff
@@ -26,7 +28,12 @@ func NewRestartCommand(dockerCli command.Cli) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "restart [OPTIONS] CONTAINER [CONTAINER...]",
 		Short: "Restart one or more containers",
-		Args:  cli.RequiresMinArgs(1),
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 && !cmd.Flags().Changed("all") {
+				return errors.New("\"docker restart\" requires at least 1 argument.")
+			}
+			return nil
```

Comment:
I wasn't totally sure how to handle this - IMO listing containers and specifying `--all` should be mutually exclusive. Alternatively, we could append the containers retrieved from the docker engine to the list of containers specified (which is currently what I'm doing) but this means that containers might be restarted twice unless we treat the list like a set and do processing on that. 

Open to suggestions, for the time being I went with the faster coding option for the sake of getting input

Reply:
It looks ok just change the message to something like:
```
"docker restart" requires at least 1 argument unless "--all" was set
```